admin/class-sitemaps-admin.php

```diff
@@ -98,4 +113,67 @@ function status_transition( $new_status, $old_status, $post ) {
 			}
 		}
 	}
+
+	/**
+	 * While bulk importing, just save unique post_types
+	 *
+	 * When importing is done, if we have a post_type that is saved in the sitemap
+	 * try to ping the search engines
+	 *
+	 * @param string   $new_status New post status.
+	 * @param string   $old_status Old post status.
+	 * @param \WP_Post $post       Post object.
+	 */
+	private function status_transition_bulk( $new_status, $old_status, $post ) {
+		if ( $new_status != 'publish' ) {
+			return;
+		}
+
+		// None of our interest..
+		if ( 'nav_menu_item' === $post->post_type ) {
+			return;
+		}
+
+		$this->importing_post_types[] = $post->post_type;
+		$this->importing_post_types   = array_unique( $this->importing_post_types );
```

Comment:
Performance improvement:
```diff
-		$this->importing_post_types[] = $post->post_type;
-		$this->importing_post_types   = array_unique( $this->importing_post_types );
+		$this->importing_post_types[$post->post_type] = $post->post_type;
```
i.e., skip the array unique and copy operation, by simply assigning the value as key, too.  That way, each value can only exist once.